hrm-domain/hrm-core/profile/profileService.ts

```diff
@@ -14,16 +14,27 @@
  * along with this program.  If not, see https://www.gnu.org/licenses/.
  */
 
-import { TResult, isErr, newErr, newOk } from '@tech-matters/types';
+import {
+  TResult,
+  newErr,
+  newOk,
+  Result,
+  isOk,
+  ErrorResult,
+  ensureRejection,
+  isErr,
+  newOkFromData,
+} from '@tech-matters/types';
 import { isFuture } from 'date-fns';
 
 import * as profileDB from './profileDataAccess';
 import { db } from '../connection-pool';
-import { txIfNotInOne } from '../sql';
+import { DatabaseErrorResult, inferPostgresErrorResult, txIfNotInOne } from '../sql';
```

Comment:
The fact that this layer needs to be aware of `DatabaseErrorResult` and `inferPostgresErrorResult` confuses me a bit, I thought this layer should be agnostic from what DB engine we use.

Reply:
DatabaseErrorResult is database agnostic. Fair point about using infer here. It just seemed like a better pinch point for creating the error result than having to add result types to 4 methods then do an `ensureRejection` etc.